doc/changelog.md

```diff
@@ -55,6 +55,9 @@
   operations and control flow operations.
   [(#353)](https://github.com/PennyLaneAI/catalyst/pull/353)
 
+* Add the `GENERATE_DEVICE_ENTRYPOINT` template macro for generating a device entry-point in the runtime plugin system.
```

Comment:
Can we change the name from `ENTRYPOINT` to something like `CONSTRUCTOR`?

Reply:
It's changed to `GENERATE_DEVICE_FACTORY`. what do you think?

Reply:
Good!

Reply:
Just curious why you prefer this @erick-xanadu ? We do refer to this function as both a device factory, but also the library entry point, in the docs.

Reply:
Well it is the point in the code where you entire the library, but maybe you're right and this would be seen as confusing since it's not the "program entry point" 🤔